internal/app/orchestrator/orchestrator.go

```diff
@@ -63,14 +61,21 @@ type orchestrator struct {
 // New instantiates the mapper, cache, upstream client components necessary for
 // the orchestrator to operate and returns an instance of the instantiated
 // orchestrator.
-func New(ctx context.Context, l log.Logger, mapper mapper.Mapper, upstreamClient upstream.Client) Orchestrator {
+func New(ctx context.Context,
+	l log.Logger,
+	mapper mapper.Mapper,
+	upstreamClient upstream.Client,
+	cacheConfig *bootstrapv1.Cache) Orchestrator {
 	orchestrator := &orchestrator{
 		logger:         l.Named(component),
 		mapper:         mapper,
 		upstreamClient: upstreamClient,
 	}
 
-	cache, err := cache.NewCache(cacheMaxEntries, orchestrator.onCacheEvicted, cacheTTL)
+	// Initialize cache.
+	cache, err := cache.NewCache(int(cacheConfig.MaxEntries),
+		orchestrator.onCacheEvicted,
+		time.Duration(cacheConfig.Ttl.Nanos)*time.Nanosecond)
```

Comment:
Same with this, any reason we don't just store the ttl as type time.Duration?

Reply:
I don't think it's possible to set the proto type to time.Duration. We could leverage gogo to convert more seamlessly between the two: https://github.com/gogo/protobuf/blob/master/extensions.md#more-canonical-go-structures.